### PR TITLE
more lintings

### PR DESCRIPTION
**Pull request by fperrad**

### CODE REVIEW

File: common-algo.c [hunk 1]
```diff
@@ -82,11 +82,9 @@ static const struct dropbear_cipher dropbear_3des =
 const struct dropbear_cipher dropbear_nocipher =
 	{NULL, 16, 8}; 
 
-/* A few void* s are required to silence warnings
- * about the symmetric_CBC vs symmetric_CTR cipher_state pointer */
 #if DROPBEAR_ENABLE_CBC_MODE
 const struct dropbear_cipher_mode dropbear_mode_cbc =
-	{(void*)cbc_start, (void*)cbc_encrypt, (void*)cbc_decrypt};
+	{cbc_start, cbc_encrypt, cbc_decrypt};
```
Comment (mkj) [inline common-algo.c]: This change isn't valid, cbc_decrypt and ctr_decrypt actually have different signatures. https://travis-ci.org/mkj/dropbear/jobs/238374212

File: common-kex.c [hunk 2]
```diff
@@ -696,7 +696,7 @@ void kexecdh_comb_key(struct kex_ecdh_param *param, buffer *pub_them,
 struct kex_curve25519_param *gen_kexcurve25519_param () {
 	/* Per http://cr.yp.to/ecdh.html */
 	struct kex_curve25519_param *param = m_malloc(sizeof(*param));
-	const unsigned char basepoint[32] = {9};
+	static const unsigned char basepoint[32] = {9};
```
Comment (mkj) [inline common-kex.c]: What's the advantage of static here?

File: crypto_desc.c [hunk 3]
```diff
@@ -13,7 +13,7 @@
  * This should be run before using any of the ciphers/hashes */
 void crypto_init() {
 
-	const struct ltc_cipher_descriptor *regciphers[] = {
+	static const struct ltc_cipher_descriptor * const regciphers[] = {
```
Comment (mkj) [inline crypto_desc.c]: Same for static here. "static" seems to make the .o bigger too.

File: curve25519-donna.c [hunk 4]
```diff
@@ -711,8 +711,6 @@ crecip(limb *out, const limb *z) {
   /* 2^255 - 21 */ fmul(out,t1,z11);
 }
 
-int curve25519_donna(u8 *, const u8 *, const u8 *);
```
Comment (mkj) [inline curve25519-donna.c]: Leave this as-is, curve25519_donna is from upstream

File: scp.c [hunk 5]
```diff
@@ -120,7 +120,7 @@ killchild(int signo)
 static int
 do_local_cmd(arglist *a)
```
Comment (mkj) [inline scp.c]: The scp files should stay unchanged, they're from OpenSSH

**Comment by fperrad** [comment]
updated with yours comments.

**Comment by mkj** [comment]
Thanks François

**Pull request merged**